static/app/components/smartSearchBar/searchDropdown.tsx

```diff
@@ -28,43 +32,87 @@ class SearchDropdown extends PureComponent<Props> {
     onClick: function () {},
   };
 
-  renderDescription = (item: SearchItem) => {
-    const searchSubstring = this.props.searchSubstring;
-    if (!searchSubstring) {
-      if (item.type === ItemType.INVALID_TAG) {
-        return (
-          <Invalid>
-            {tct("The field [field] isn't supported here. ", {
-              field: <strong>{item.desc}</strong>,
-            })}
-            {tct('[highlight:See all searchable properties in the docs.]', {
-              highlight: <Highlight />,
-            })}
-          </Invalid>
-        );
-      }
-
-      return item.desc;
+  renderItemTitle = (item: SearchItem, isChild?: boolean) => {
+    if (!item.title) {
+      return null;
     }
 
-    const text = item.desc;
+    const fullWord = item.title;
 
-    if (!text) {
-      return null;
-    }
+    const words =
+      item.kind !== FieldValueKind.FUNCTION ? fullWord.split('.') : [fullWord];
+    const [firstWord, ...restWords] = words;
+    const isFirstWordHidden = isChild;
+
+    const combinedRestWords = restWords.length > 0 ? restWords.join('.') : null;
+
+    const searchSubstring = this.props.searchSubstring;
 
-    const idx = text.toLowerCase().indexOf(searchSubstring.toLowerCase());
+    if (searchSubstring) {
+      const idx =
+        restWords.length === 0
+          ? fullWord.toLowerCase().indexOf(searchSubstring.split('.')[0])
+          : fullWord.toLowerCase().indexOf(searchSubstring);
+
+      // Below is the logic to make the current query bold inside the result.
+      if (idx !== -1) {
```

Comment:
In general, I'd prefer to extract into functions rather than create a variable and mutate it later on. Makes for a more readable codebase. So instead of declaring `renderedRest` and modifying it in various cases, extract it to its own function so you can do `renderedRest = makeRenderedRest({ .. })`

Or perhaps you could make it its own component? I think this `renderItemTitle` function would be better written as a component (and rending subcomponents that hold the logic seen here).